### PR TITLE
Fix duplication bug in `User.donors`

### PR DESCRIPTION
**Pull request by seanpdoyle**
Add coverage for a bug uncovered when a donor has made a donation.

### CODE REVIEW

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -59,6 +59,7 @@
       create(:donor, admin: true, name: "Admin/Donor")
       create(:admin, name: "Admin")
       create(:cyclist, name: "Cyclist")
+      create(:donation, donor: donor)
```
Comment (jferris) [inline spec/models/user_spec.rb]: I'm confused as to why this change to the test covers the corresponding change to the model.

Comment (seanpdoyle) [inline spec/models/user_spec.rb]: @jferris right, sorry for pulling you into this.

A `Donation` has a `has_one :donor, through: :location` relationship. The factory should be called with a `location`, not a `donor`.
